src/nbl/video/CVulkanCommandBuffer.h

```diff
@@ -39,8 +39,25 @@ class CVulkanCommandBuffer : public IGPUCommandBuffer
         freeSpaceInCmdPool();
     }
 
-    bool begin(core::bitflag<E_USAGE> recordingFlags, const SInheritanceInfo* inheritanceInfo=nullptr) override
+    void resetCommon() override
     {
+        freeSpaceInCmdPool();
+        IGPUCommandBuffer::resetCommon();
+    }
+
+    bool begin(core::bitflag<E_USAGE> recordingFlags, const SInheritanceInfo* inheritanceInfo = nullptr) override
+    {
+        // Resources are freed when the command buffer is reset, so:
+        // - When ECF_RESET_COMMAND_BUFFER_BIT is set on the command pool and we are resetting the command buffer
+        //     - Command buffer is reset when calling begin or reset (handled elsewhere)
+        // - When ECF_RESET_COMMAND_BUFFER_BIT is NOT set on the command pool, and the pool has been reset
+        if (m_cmdpool->getCreationFlags().hasFlags(video::IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT))
+        {
+            // "reset" the command buffer by removing existing resources
+            if (m_state != ES_PENDING) freeSpaceInCmdPool();
```

Comment:
use `canReset` functions instead of these two if conditions it's exactly equivalent and makes sense